vllm/worker/hpu_model_runner.py

```diff
@@ -2417,27 +2422,47 @@ def warmup_scenario(self,
                             context_size=seq_len if is_prompt else 1,
                             dtype=self.model_config.dtype,
                             device=self.device)
-                self.execute_model(inputs,
-                                   kv_caches,
-                                   intermediate_tensors=intermediate_tensors,
-                                   warmup_mode=True)
+                output = self.execute_model(
+                    inputs,
+                    kv_caches,
+                    intermediate_tensors=intermediate_tensors,
+                    warmup_mode=True)
+                if deferred_sampling:
+                    self.execute_sample(
+                        hidden_states=output,
+                        model_input=inputs,
+                        num_steps=1,
+                    )
             else:  # decode with multi-step
                 inputs = dataclasses.replace(inputs,
                                              is_first_multi_step=True,
                                              is_last_step=False)
-                self.execute_model(inputs,
-                                   kv_caches,
-                                   warmup_mode=True,
-                                   num_steps=2,
-                                   seqs=seqs)
+                output = self.execute_model(inputs,
+                                            kv_caches,
+                                            warmup_mode=True,
+                                            num_steps=2,
+                                            seqs=seqs)
+
+                if deferred_sampling:
+                    self.execute_sample(
+                        hidden_states=output,
+                        model_input=inputs,
+                        num_steps=2,
```

Comment:
current implementation of execute_sample seems only support num_steps=1
"
assert num_steps == 1
"